Source/RapyutaSimulationPlugins/Public/Core/RRGeneralUtils.h

```diff
@@ -294,4 +294,56 @@ class RAPYUTASIMULATIONPLUGINS_API URRGeneralUtils : public UBlueprintFunctionLi
         OutValue = InDefaultValue;
         return false;
     }
+
+    /**
+     * @brief Check if a velocity's magnitude exceeds a given max speed
+     *
+     * @param InVel
+     * @param InMaxSpeed
+     * @param b2DMovement
+     */
+    FORCEINLINE static bool IsVelocityExceedingMaxSpeed(const FVector& InVel, float InMaxSpeed, bool b2DMovement)
```

Comment:
Can we genelize this as IsVecotorExceedMax?

Reply:
I also thought about it as a util in `URRMathUtils` instead. If we do so, since it would be not very nice as writing a `foo(FVector, MaxMagnitude, b2DVector)` so it should be a template-based one like:
```
template<typename T>
static bool IsVectorExceeedingMaxMagnitude(const T& InVector, float InMaxMagnitude)
```
and call UE's built-in `SizeSquared()` for either T being `FVector` or `FVector2D`.
However, there are 2 issues with it:
* Using it would make us have to create an automatic var `FVector2D` from the input `Velocity`, before passing it to the util
* `b2DMovement` cannot be a parameter (due to its already inferred from typename T)

So yes, we could but then the util itself would not be very essential since `Velocity` is inherently always a 3D `FVector`. I haven't found a better idea yet tho.

Reply:
I don't think you need to make it template but just for FVector.